schism/compiler.ss

```diff
@@ -873,10 +878,13 @@
         `(lambda ,(cadr expr) ,(simplify-expr (caddr expr))))
        (else
         (trace-and-error expr 'simplify-expr "unrecognized expr")))))
-  (define (simplify-function fn)
-    `(,(car fn) ,(simplify-expr (cadr fn))))
-  (define (simplify-functions functions)
-    (map simplify-function functions))
+  (define (simplify-definition def)
+    (let ((tag (car def)))
+      (if (eq? tag 'define)
+          `(define ,(cadr def) ,(simplify-expr (caddr def)))
+          `(,(car def) ,(simplify-expr (cadr def))))))
```

Comment:
Can you use `tag` here instead of `(car def)`?

Reply:
Ah of course! Sorry, I haven't yet passed through the changes making sure everything's clean, since I assumed there were still some issues to resolve